File: sdks/python/bolt_prediction_sdk/models/tenant_api_key_create_request.py
"""
TenantAPIKeyCreateRequest 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class TenantApiKeyCreateRequest(SDKBaseModel):
    """TenantAPIKeyCreateRequest"""

    key_name: Optional[Any] = Field(description="密钥名称", default=None)
    permissions: Optional[Any] = Field(description="权限列表", default=None)
    rate_limit: Optional[int] = Field(description="速率限制 每分钟", default=1000)
    user_id: Optional[Any] = Field(description="关联用户ID", default=None)
    expires_at: Optional[Any] = Field(description="过期时间", default=None)

File: sdks/java/src/main/java/com/boltprediction/sdk/model/TenantApiKeyCreateRequest.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** TenantAPIKeyCreateRequest */
@JsonIgnoreProperties(ignoreUnknown = true)
public class TenantApiKeyCreateRequest {

    @JsonProperty("key_name")
    private Object keyName;

    @JsonProperty("permissions")
    private Object permissions;

    @JsonProperty("rate_limit")
    private Integer rateLimit;

    @JsonProperty("user_id")
    private Object userId;

    @JsonProperty("expires_at")
    private Object expiresAt;

    public Object getKeyName() {
        return keyName;
    }

    public void setKeyName(Object keyName) {
        this.keyName = keyName;
    }

    public Object getPermissions() {
        return permissions;
    }

    public void setPermissions(Object permissions) {
        this.permissions = permissions;
    }

    public Integer getRateLimit() {
        return rateLimit;
    }

    public void setRateLimit(Integer rateLimit) {
        this.rateLimit = rateLimit;
    }

    public Object getUserId() {
        return userId;
    }

    public void setUserId(Object userId) {
        this.userId = userId;
    }

    public Object getExpiresAt() {
        return expiresAt;
    }

    public void setExpiresAt(Object expiresAt) {
        this.expiresAt = expiresAt;
    }

}